checkisjson_test.go

```diff
@@ -0,0 +1,453 @@
+package httprequest_test
+
+import (
+	"net/http"
+	"strings"
+
+	gc "gopkg.in/check.v1"
+
+	"github.com/juju/httprequest"
+)
+
+type checkIsJSONSuite struct{}
+
+var _ = gc.Suite(&checkIsJSONSuite{})
+
+var checkIsJSONTests = []struct {
+	about       string
+	contentType string
+	body        string
+	expectError string
+}{{
+	about:       "simple json",
+	contentType: "application/json; charset=UTF-8",
+}, {
+	about:       "simple json with charset",
+	contentType: "application/json; charset=UTF-8",
+}, {
+	about:       "plain text",
+	contentType: "text/plain; charset=UTF-8",
+	body:        "   some\n   text\t\n",
+	expectError: `unexpected content type text/plain; want application/json; content: some; text`,
+}, {
+	about:       "plain text with leading newline",
+	contentType: "text/plain; charset=UTF-8",
+	body:        "\nsome text",
+	expectError: `unexpected content type text/plain; want application/json; content: some text`,
+}, {
+	about:       "unknown content type",
+	contentType: "something",
+	body:        `some stuff`,
+	expectError: `unexpected content type something; want application/json; content: "some stuff"`,
```

Comment:
I'd demonstrate that some stuff newlines or spaces are not modified in this case.